src/rapids_pre_commit_hooks/alpha_spec.py

```diff
@@ -111,42 +111,52 @@ def create_specifier_string(specifiers):
         if req.name in RAPIDS_ALPHA_SPEC_PACKAGES or is_rapids_cuda_suffixed_package(
             req.name
         ):
-            has_alpha_spec = any(str(s) == ALPHA_SPECIFIER for s in req.specifier)
-            if args.mode == "development" and not has_alpha_spec:
-                linter.add_warning(
-                    (node.start_mark.index, node.end_mark.index),
-                    f"add alpha spec for RAPIDS package {req.name}",
-                ).add_replacement(
-                    (node.start_mark.index, node.end_mark.index),
-                    str(
-                        req.name
-                        + create_specifier_string(
-                            {str(s) for s in req.specifier} | {ALPHA_SPECIFIER}
-                        )
-                    ),
-                )
-            elif args.mode == "release" and has_alpha_spec:
-                linter.add_warning(
-                    (node.start_mark.index, node.end_mark.index),
-                    f"remove alpha spec for RAPIDS package {req.name}",
-                ).add_replacement(
-                    (node.start_mark.index, node.end_mark.index),
-                    str(
-                        req.name
-                        + create_specifier_string(
-                            {str(s) for s in req.specifier} - {ALPHA_SPECIFIER}
-                        )
-                    ),
-                )
-
-
-def check_packages(linter, args, node):
+            anchor = (
+                keys[0]
+                if (keys := [key for key, value in anchors.items() if value == node])
```

Comment:
We probably don't need to materialize `keys` as a list just to get the first item. This should exit early. Perhaps try something like this:

```python
for key, value in anchors.items():
    if value == node:
        anchor = key
        break
else:
    anchor = None
```
or
```python
anchor = next((key for key, value in anchors.items() if value == node), None)
```